### PR TITLE
Add a rename function.

### PR DESCRIPTION
**Pull request by mattgrande**
As suggested in #2089; Code modified from the [cookbook](https://github.com/ramda/ramda/wiki/Cookbook#rename-keys-of-an-object).

**Edit:** I missed that there was a linter (should've checked, my apologies). Consider adding that step to `CONTRIBUTING.md`?

### CODE REVIEW

**Comment by adispring** [comment]
I just modified the implement of `rename` more neat. https://github.com/ramda/ramda/wiki/Cookbook#rename-keys-of-an-object

**Comment by davidchambers** [comment]
@adispring, while you're at it you should remove the incorrect use of `||` in `keysMap[key] || key`. This will not work correctly if the left-hand operand evaluates to `''`.

File: src/rename.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Creates a new object with the own properties of the provided object, but the
+ * keys renamed according to the keysMap object as `{oldKey: newKey}`.
+ * When some key is not found in the keysMap, then it's passed as-is.
+ *
+ * @sig {a: b} -> {a: *} -> {b: *}
+ */
+module.exports = _curry2(function rename(keysMap, obj) {
+  return reduce(function(acc, key) {
+    acc[keysMap[key] || key] = obj[key];
```
Comment (davidchambers) [inline src/rename.js]: We cannot use `||` here, as the left-hand side may evaluate to `''`.

Comment (mattgrande) [inline src/rename.js]: Huh, I didn't realise that `''` would be a valid key for an object... Although I guess it makes sense.

I'll replace this with a null/undefined check.

Comment (davidchambers) [inline src/rename.js]: > I'll replace this with a null/undefined check.

That would be problematic too, due to inherited properties:

```javascript
> ({})['toString'] || 'toString'
function toString() { [native code] }
```

Using [`R.has`][1] might be a good idea.


[1]: http://ramdajs.com/0.23.0/docs/#has

**Comment by CrossEye** [comment]
> Sorry, null and undefined can also be valid Object keys
> (edit: ignore that, they just coerce to the strings 'null' and 'undefined')

It's an interesting philosophical point though.  :smile:

Because they coerce also on a get, it's only because of things like `Object.keys` that you can even tell that they're not really allowed.  Thus:

```js
const obj = {};
obj[null]; //=> undefined
null in obj; //=> false

obj[null] = 42;
obj[null]; //=> 42
null in obj; //=> true
```

You can tell that it's not actually a property by noting that 

```js
Object.keys(obj); //=> ["null"]
```

But that's fairly subtle.

**Comment by mattgrande** [comment]
Updated; Using `has` instead of null check. Sorry for the delay.

**Comment by adrian-gierakowski** [comment]
I have used exactly this kind of function quite a few times, however I think it would be more beneficial to add more general `mapKeys` function ( like https://lodash.com/docs/4.17.4#mapKeys ). `mapKeys` could work with functions which accept only one arg ( key ) and `mapKeysWithValue` could take a function which accepts both `key` and `value` ( o `value` and `key` ). `rename` could then be expressed in therms of one of those

**Comment by CrossEye** [comment]
@adrian-gierakowski:

Although a `mapKeys` function (and/or a `mapKeysWithValue` one) might be useful, I don't think they'd take the place of this one.  While such *could* be used to create `renameKeys` behavior, it would not be particularly comfortable for a user to do so, and it seems a reasonably common request.

So if we do `mapKeys`, let's let it stand separate from this one.

**Comment by mattgrande** [comment]
Trying to clear out some long-open PRs. Do people still have interest in this one, or should I close it?

**Comment by CrossEye** [comment]
I have no strong feelings either way.  Others in @ramda/core ?

**Comment by mattgrande** [comment]
Seems like there was never much demand for this; Closing.

**Comment by rdsedmundo** [comment]
I wish we had this. :)

**Pull request closed without merging**